core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -77,6 +77,9 @@ abstract class AbstractFetcherThread(name: String,
       partitionMapCond.signalAll()
     }
     awaitShutdown()
+
+    fetcherStats.unregister()
+    fetcherLagStats.unregister()
```

Comment:
Seems like it is. I put it after `awaitShutdown` so that there wouldn't be any race conditions with metric creation in the thread itself. The main concern is whether another fetcher could be added at the same time with the same name, but it looks like `AbstractFetcherManager` guards fetcher creation/shutdown with its own lock.